drivers/volume/kdmp/kdmp.go

```diff
@@ -0,0 +1,300 @@
+	// Try to get info from the PV since storage class could be deleted
+	pv, err := coreOps.GetPersistentVolume(pvc.Spec.VolumeName)
+	if err != nil {
+		log.PVCLog(pvc).Warnf("error getting pv %v for pvc %v: %v", pvc.Spec.VolumeName, pvc.Name, err)
+		return false
+	}
+	// for px proxy volume use kdmp by default
+	storageClassName := k8shelper.GetPersistentVolumeClaimClass(pvc)
+	if storageClassName != "" {
+		storageClass, err := storage.Instance().GetStorageClass(storageClassName)
+		if err != nil {
+			logrus.Warnf("Error getting storageclass %v for pvc %v: %v", storageClassName, pvc.Name, err)
+		}
+		if _, ok := storageClass.Parameters[proxyEndpoint]; ok {
+			return true
+		}
+	}
+	return k.OwnsPV(pv)
+}
+
+func (k *kdmp) OwnsPV(pv *v1.PersistentVolume) bool {
+
+	// for csi volumes (with or without snapshot) use kdmp driver by default
+	// except for px csi volumes
+	if pv.Spec.CSI != nil {
+		// We px csi volume fall back to default csi volume driver
+		if pv.Spec.CSI.Driver == snapv1.PortworxCsiProvisionerName ||
+			pv.Spec.CSI.Driver == snapv1.PortworxCsiDeprecatedProvisionerName {
+			return false
+		}
+		log.PVLog(pv).Tracef("CSI Owns PV: %s", pv.Name)
+		return true
+	}
+
+	return false
+}
+
+func (k *kdmp) StartBackup(backup *storkapi.ApplicationBackup,
+	pvcs []v1.PersistentVolumeClaim,
+) ([]*storkapi.ApplicationBackupVolumeInfo, error) {
+	log.ApplicationBackupLog(backup).Debugf("started kdmp backup: %v", backup.Name)
+	volumeInfos := make([]*storkapi.ApplicationBackupVolumeInfo, 0)
+	for _, pvc := range pvcs {
+		if pvc.DeletionTimestamp != nil {
+			log.ApplicationBackupLog(backup).Warnf("Ignoring PVC %v which is being deleted", pvc.Name)
+			continue
+		}
+		volumeInfo := &storkapi.ApplicationBackupVolumeInfo{}
+		volumeInfo.PersistentVolumeClaim = pvc.Name
+		volumeInfo.Namespace = pvc.Namespace
+		volumeInfo.DriverName = driverName
+		volume, err := core.Instance().GetVolumeForPersistentVolumeClaim(&pvc)
+		if err != nil {
+			return nil, fmt.Errorf("error getting volume for PVC: %v", err)
+		}
+		volumeInfo.Volume = volume
+		// create kdmp cr
+
+		dataExport := &kdmpapi.DataExport{}
+		dataExport.Name = "stork-" + pvc.Name + "-" + pvc.Namespace
+		dataExport.Namespace = pvc.Namespace
+		// TODO: this needs to be generic, maybe read it from configmap
+		dataExport.Spec.Type = kdmpapi.DataExportKopia
+		dataExport.Spec.Destination = kdmpapi.DataExportObjectReference{
+			Kind:       reflect.TypeOf(storkapi.BackupLocation{}).Name(),
+			Name:       backup.Spec.BackupLocation,
+			Namespace:  backup.Namespace,
+			APIVersion: StorkAPIVersion,
+		}
+		dataExport.Spec.Source = kdmpapi.DataExportObjectReference{
+			Kind:       pvc.Kind,
+			Name:       pvc.Name,
+			Namespace:  pvc.Namespace,
+			APIVersion: pvc.APIVersion,
+		}
+		logrus.Debugf("Source for KDMP: %v", dataExport.Spec.Source)
+		logrus.Debugf("Destination for KDMP: %v", dataExport.Spec.Destination)
+		// TODO: do we need retries here
```

Comment:
With the change to job framework model in kdmp limiting the number of jobs, do we still need this? Jobs will not even start if the threshold has not been reached.

Nit: typo `excedded`